bitcoin/src/pow.rs

```diff
@@ -924,9 +926,137 @@ impl kani::Arbitrary for U256 {
     }
 }
 
+/// Gets the target for the block after `current_header`.
+///
+/// Implements the `GetNextWorkRequired` function from Bitcoin core.
+///
+/// # Args
+///
+/// `new_block_timestamp` is only used on testnet.
+///
+/// > Special difficulty rule for testnet: If the new block's timestamp is more
+/// > than 2*10 minutes then allow mining of a min-difficulty block.
+pub fn next_target<F, E>(
+    params: &Params,
+    current_height: absolute::Height,
+    current_header: block::Header,
+    new_block_timestamp: u32, // Only used on testnet for special difficulty rule.
```

Comment:
it is better put this inside the `params`? and maybe have this as the default value for Testnet params?

Reply:
I don't understand this comment sorry. `new_block_timestamp` is the timestamp of the next block, ie, `current_height + 1`. I don't see what you mean to put in `Params`.

Reply:
I made it an `Option` and added an assertion as well as documented the panic.
```rust
    if params.allow_min_difficulty_blocks {
        assert!(next_block_timestamp.is_some())
    }
``